src/mmw/apps/modeling/mapshed/calcs.py

```diff
@@ -213,6 +213,47 @@ def manure_spread(aeu):
     return [n_spread] * num_land_uses, [p_spread] * num_land_uses
 
 
+def ag_ls_c_p(geom):
+    """
+    Given a geometry, calculates the area-weighted average value of LS, C, and
+    P factors for agriculatural land use tyeps within the geometry, namely
+    Hay/Pasture and Cropland.
+    """
+    sql = '''
+          WITH clipped_counties AS (
+              SELECT ST_Intersection(geom,
+                                     ST_SetSRID(ST_GeomFromText(%s),
+                                                4326)) AS geom_clipped,
+                     ms_county_animals.*
+              FROM ms_county_animals
+              WHERE ST_Intersects(geom,
+                                  ST_SetSRID(ST_GeomFromText(%s),
+                                             4326))
+          ), clipped_counties_with_area AS (
+              SELECT ST_Area(geom_clipped) /
+                     ST_Area(ST_SetSRID(ST_GeomFromText(%s),
+                                        4326)) AS clip_percent,
+                     clipped_counties.*
+              FROM clipped_counties
+          )
+          SELECT SUM(hp_ls * clip_percent) AS hp_ls,
+                 SUM(hp_c * clip_percent) AS hp_c,
+                 SUM(hp_p * clip_percent) AS hp_p,
+                 SUM(crop_ls * clip_percent) AS crop_ls,
+                 SUM(crop_c * clip_percent) AS crop_c,
+                 SUM(crop_p * clip_percent) AS crop_p
+          FROM clipped_counties_with_area;
+          '''
+
+    with connection.cursor() as cursor:
+        cursor.execute(sql, [geom.wkt, geom.wkt, geom.wkt])
+
+        ag_lscp = namedtuple('Ag_LS_C_P',
+                             [col[0] for col in cursor.description])
+
+        return ag_lscp(*(cursor.fetchone()))
```

Comment:
They are assigned here. The `namedtuple` line above makes essentially a `struct` with the field names corresponding to query, and this line populates them with the right values.

I did this rather than make a dictionary because I preferred to use `ag_lscp.crop_c` rather than `ag_lscp['crop_c']`. We do the same thing with [`WeatherStation`](https://github.com/WikiWatershed/model-my-watershed/blob/41d94d26e51502fa723eadb9edba8585e32a244a/src/mmw/apps/modeling/mapshed/calcs.py#L70-L73) above.